lib/hawkular/alerts/alerts_api.rb

```diff
@@ -210,6 +210,22 @@ def list_events(*criteria)
       query = generate_query_params(*criteria)
       http_get('/events' + query).map { |e| Event.new(e) }
     end
+
+    # Inject an event into Hawkular-alerts
+    # @param [String] id Id of the event must be unique
+    # @param [String] category Event category for further distinction
+    # @param [String] text Some text to the user
+    # @param [Hash<String,Object>] context Additional context information.
+    def create_event(id, category, text, context = {})
+      event = {}
+      event['id'] = id
+      event['ctime'] = Time.now.to_i * 1000
+      event['category'] = category
+      event['text'] = text
+      event['context'] = context unless context.nil?
+
+      http_post('/events', event)
```

Comment:
On 3 May 2016, at 15:01, abonas wrote:

> why are events needs to be injected via Ruby to hawkular side?

There is no immediate need from the MiQ side, but as this is a general 
client gem,
other users may need it.

It certainly allows some testing of event catching in MiQ from within 
the Ruby code,
where the event is first injected into Hawkular and then after a bit of 
wait time
retrieved by the event catchers.